EVENT.md

```diff
@@ -603,6 +604,66 @@ Set-Cookie: ar_debug=1; SameSite=None; Secure; HttpOnly
 ```
 If a cookie of this form is not present, debugging information will be ignored.
 
+### Optional: error reports
+
+We also introduce a non-cookie based debugging framework to allow developers to
+monitor certain failures in the attribution registrations.
+
+The browser will send error reports in the following source registration failure
+modes:
+
+* a source is rejected due to the [reporting origin
+  limits](#reporting-origin-limits)
+* a source is rejected due to the [destination
+  limits](#limiting-the-number-of-unique-destinations-covered-by-pending-sources)
+
+The browser will send non-credentialed secure HTTP POST requests to the
+reporting endpoints, see [below](#reporting-endpoints). The report data is
+included in the request body as a JSON object:
+```jsonc
+{
+  "type": "<report type>", // could be "unattributed-reporting-origin-limit" or
+  "source-destination-limit"
+  "body": {
+    "limit": 100, // a constant integer
+    "source_event_id": "<source event id in the source registration>",
+    "source_site": "https://source.example",
+    "destination_site": "https://destination.example",
+    "reporting_origin": "https://reporting.example.com"
+  }
+}
+```
+
+These error reports will be sent immediately upon the error occuring during
+source registration.
+
+Note that if other failures are reported using this framework, the browser may
+enforce a random small delay and drop some data (e.g. `source_event_id`) to
+avoid the page being identifiable.
+
+#### Reporting endpoints
+
+The reporting origins may opt in to receive error reports by adding a new
+`error_reporting` dictionary field to the `Attribution-Reporting-Register-Source` header:
+```jsonc
+{
+  ... // exsiting fields
+
+  "error_reporting": true // default to false if not present
+}
+```
+
+The error reports will be sent to a new endpoint:
+```
+https://<reporting origin>/.well-known/attribution-reporting/error
+```
+
+The top-level sites may opt in to receive error reports by declaring a reporting
+endpoint via a HTTP header in the top-level document response:
+```
+Attribution-Error-Reporting-Endpoint: <reporting endpoint>
```

Comment:
Noted that it can be an arbitrary URL.

It looks like the Reporting API doesn't require the reporting endpoint to be same-origin. Do we need this restriction?

Reply:
Chiming in from a developer experience standpoint: unless there's a security/privacy reason to mandate same-origin, there's value in this URL being potentially arbitrary. This opens the possibility for developers to leverage third-party error monitoring services, as it's the case today for Reporting API. While these services don't exist yet for Attribution Reporting, they may in the future.